README.md

```diff
@@ -1,2 +1,32 @@
 # comp110-worksheet-5
 Base repository for COMP110 worksheet 5
+
+a. This algorithm creates a list to check duplicate in list by declaring N to be the length and creates a
+loop to check if i equals zero to N then checks with another loop if j
+equals zero to N then if it does it will see if i is not equal to j list
+then it will return true if it does have duplicate in list and if not it will return false.
+
+b. The worst case running time of the algorithm is quadratic because it runs two loops for listing i and j
+ inside the code.
+
+c. The algorithm is still correct because it still runs because it checks the duplicates throughout both 
+lists and still returns true if there isn't any duplicates and returns false if there is.
+
+d. The algorithm will run twice as fast because it is a quadratic algorithm and as stated above it includes two 
+lists that will loop through the code
+
+e. Yes the time complexity of the algorithm is still quadratic as quadratic time complexity represents an 
+algorithm whose performance is directly proportional to the squared size of the input data set as the two lists.
+
+f. O(n log n)
+
+g. O(n) - I think that this is the time complexity of the 2nd algorithm
+in big O notation because it looks like a linear algorithm as the 1st algorithm is quadratic and the time complexity
+notation looks like this "O(n²)" as it includes two for loops whereas the 2nd only includes one.
```

Comment:
We also have to take account of the sort that takes place before the search -- so the overall time complexity is `O(n log n + n)`, which simplifies to `O(n log n)`.